In [2]:
import pandas as pd
import os
import numpy as np
from os import getcwd
getcwd()

'/Users/damola/Desktop/GA/DS-SF-24/HW Assignments/HW1'

In [3]:
os.chdir('/Users/damola/Desktop/GA/DS-SF-24/HW Assignments/')
df=pd.read_csv('~/Desktop/GA/DS-SF-24/HW Assignments/HW1/titanic.csv')

In [4]:
print df.head()

   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  


In [5]:
print df.describe()

       PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200  


1. How many passengers are in our passenger list?

There are 891 passengers in the dataset

2. What is the overall survival rate?

It was 38.4% (from mean survived)

In [9]:
print df.groupby(['Sex']).describe()

                     Age        Fare       Parch  PassengerId      Pclass  \
Sex                                                                         
female count  261.000000  314.000000  314.000000   314.000000  314.000000   
       mean    27.915709   44.479818    0.649682   431.028662    2.159236   
       std     14.110146   57.997698    1.022846   256.846324    0.857290   
       min      0.750000    6.750000    0.000000     2.000000    1.000000   
       25%     18.000000   12.071875    0.000000   231.750000    1.000000   
       50%     27.000000   23.000000    0.000000   414.500000    2.000000   
       75%     37.000000   55.000000    1.000000   641.250000    3.000000   
       max     63.000000  512.329200    6.000000   889.000000    3.000000   
male   count  453.000000  577.000000  577.000000   577.000000  577.000000   
       mean    30.726645   25.523893    0.235702   454.147314    2.389948   
       std     14.678201   43.138263    0.612294   257.486139    0.813580   

3. How many male passengers were onboard?

There were 577 male passengers onboard

4. How many female passengers were onboard?

There were 314 female passengers onboard

5. What is the overall survival rate of male passengers?

The overall survival rate of male passengers was 18.9%

6. What is the overall survival rate of female passengers?

The overall survival rate of female passengers was 74.2%

In [33]:
df2 = df[df.Age > 0.1]

In [34]:
df2.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [35]:
df2.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   714.000000  714.000000  714.000000  714.000000  714.000000   
mean    448.582633    0.406162    2.236695   29.699118    0.512605   
std     259.119524    0.491460    0.838250   14.526497    0.929783   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     222.250000    0.000000    1.000000   20.125000    0.000000   
50%     445.000000    0.000000    2.000000   28.000000    0.000000   
75%     677.750000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    5.000000   

            Parch        Fare  
count  714.000000  714.000000  
mean     0.431373   34.694514  
std      0.853289   52.918930  
min      0.000000    0.000000  
25%      0.000000    8.050000  
50%      0.000000   15.741700  
75%      1.000000   33.375000  
max      6.000000  512.329200

7. What is the average age of all passengers onboard? 

* How did you calculate this average age?
* Note that some of the passengers do not have an age value. How did you deal with this? What are some other ways of dealing with this?

Average age is 29.7 yrs old. 

Calculated for only those who had age values assigned to them. I dealt with missings by only looking at the subset of passengers who have age values assigned. You could also deal with missing values by filling them with average values. Or by estimating age based on what you know about the individual subjects in your dataset.  

In [38]:
print df2.groupby(['Survived']).mean()

          PassengerId    Pclass        Age     SibSp     Parch       Fare
Survived                                                                 
0          442.299528  2.485849  30.626179  0.525943  0.365566  22.965456
1          457.768966  1.872414  28.343690  0.493103  0.527586  51.843205


8. What is the average age of passengers who survived?

Average age of passengers who survived was 28.34

9. What is the average age of passengers who did not survive?

Average of passengers who didn't survive was 30.62

10. At this (early) point in our analysis, what might you infer about any patterns you are seeing?

~62% of passengers were killed. Passengers who survived tended to be slightly younger (on average) than those who did not survive, and they were disproportionately female.  

11. How many passengers are in each of the three classes of service (e.g. First,
Second, Third?)

In [39]:
print df.groupby(['Pclass']).count()

        PassengerId  Survived  Name  Sex  Age  SibSp  Parch  Ticket  Fare  \
Pclass                                                                      
1               216       216   216  216  186    216    216     216   216   
2               184       184   184  184  173    184    184     184   184   
3               491       491   491  491  355    491    491     491   491   

        Cabin  Embarked  
Pclass                   
1         176       214  
2          16       184  
3          12       491  


Pclass 1 = 216, Pclass2 = 184, Pclass3 = 491 passengers

12. What is the survival rate for passengers in each of the three classes of service?

In [40]:
print df.groupby(['Pclass']).mean()

        PassengerId  Survived        Age     SibSp     Parch       Fare
Pclass                                                                 
1        461.597222  0.629630  38.233441  0.416667  0.356481  84.154687
2        445.956522  0.472826  29.877630  0.402174  0.380435  20.662183
3        439.154786  0.242363  25.140620  0.615071  0.393075  13.675550


Survival rates were 63%, 47%, and 24%, for classes 1, 2, and 3, respectively

13. What else might you conclude?

It looks like the more money you spent on a ticket fare (the higher your fare class), the more likely you were to survive. 

14. Last, if we were to build a predictive model, which features in the data do you think we should include in the model and which can we leave out? Why?

If we were building a predictive model, I'd choose survival as the outcome metric to predict. I'd definitely include fare class and sex because they seem to be really good predictors for survival based on this early analysis. 

Age also seems to be an important predictor. You're more likely to have survived with a pclass of 1 and the average age of survivors was 29, but the average age of passengers with a fare class of 1 was higher at 38.2. So there's something interesting going on with age vs. pclass, and I'd like to explore that. 

I wouldn't bother to include sibsp as it doesn't seem particluarly relevant. 